### PR TITLE
Implement _mm_hadd_pi32

### PR DESCRIPTION
**Pull request by HowJMay**
Add overloaded function, 'validateInt32' for testing datatype '__m64'.

### CODE REVIEW

File: tests/common.cpp [hunk 1]
```diff
@@ -103,6 +103,14 @@ result_t validateUInt16(__m128i a,
     return TEST_SUCCESS;
 }
 
+result_t validateInt32(__m64 a, int32_t u0, int32_t u1)
```
Comment (jserv) [inline tests/common.cpp:106]: Can you check line 38? There is already one `validateInt32`.

Comment (HowJMay) [reply tests/common.cpp:106]: But `validateInt32` at line 38 takes `__m128i` as input argument. The new `validateInt32` implemented here takes `__m64` as input argument.

Comment (jserv) [reply tests/common.cpp:106]: It was intended to add an overloading function. You shall mention in git commit message(s) as well.

**Review by jserv — CHANGES_REQUESTED** [review]
Change tests/impl.h for this intrinsic.

**Comment by jserv** [comment]
Function `result_t validateInt32(__m64 a, int32_t u0, int32_t u1)` is an "overloaded" function instead of "reload."
See https://en.wikipedia.org/wiki/Overloading

**Review by marktwtn — APPROVED** [review]

**Pull request merged**